clients/src/test/java/org/apache/kafka/clients/FetchSessionHandlerTest.java

```diff
@@ -443,17 +443,50 @@ public void testIdUsageRevokedOnIdDowngrade() {
 
             // Try to remove a topic ID from an existing topic partition (0) or add a new topic partition (1) without an ID.
             FetchSessionHandler.Builder builder2 = handler.newBuilder();
-            builder2.add(new TopicPartition("foo", 0), Uuid.ZERO_UUID,
+            builder2.add(new TopicPartition("foo", partition), Uuid.ZERO_UUID,
                     new FetchRequest.PartitionData(10, 110, 210, Optional.empty()));
             FetchSessionHandler.FetchRequestData data2 = builder2.build();
             // Should have the same session ID and next epoch, but can no longer use topic IDs.
             // The receiving broker will close the session if we were previously using topic IDs.
             assertEquals(123, data2.metadata().sessionId(), "Did not use same session when " + testType);
-            assertEquals(1, data2.metadata().epoch(), "Did not close session when " + testType);
+            assertEquals(1, data2.metadata().epoch(), "Did not have correct epoch when " + testType);
             assertFalse(data2.canUseTopicIds());
         });
     }
 
+    @Test
+    public void testIdUsageWithAllForgottenPartitions() {
+        // We want to test when all topics are removed from the session
+        List<Boolean> useTopicIdsTests = Arrays.asList(true, false);
+        useTopicIdsTests.forEach(useTopicIds -> {
+            Uuid topicId = useTopicIds ? Uuid.randomUuid() : Uuid.ZERO_UUID;
+            Short respVer = useTopicIds ? ApiKeys.FETCH.latestVersion() : 12;
+            Map<String, Uuid> topicIds = Collections.singletonMap("foo", topicId);
+            FetchSessionHandler handler = new FetchSessionHandler(LOG_CONTEXT, 1);
+            FetchSessionHandler.Builder builder = handler.newBuilder();
```

Comment:
nit: Could we add an empty line and a small comment before this `builder` similarly to what you did before `builder2`? It would make reading the test case a little easier.